sympy/integrals/manualintegrate.py

```diff
@@ -859,6 +861,24 @@ def root_mul_rule(integral):
     if next_step:
         return URule(u, u_func, None, next_step, integrand, symbol)
 
+
+def hyperbolic_rule(integral: tuple[Expr, Symbol]):
+    integrand, symbol = integral
+    if isinstance(integrand, HyperbolicFunction) and integrand.args[0] == symbol:
+        if integrand.func == sinh:
+            return HyperbolicRule('sinh', symbol, integrand, symbol)
+        if integrand.func == cosh:
+            return HyperbolicRule('cosh', symbol, integrand, symbol)
+        if integrand.func == tanh:
+            rewritten = sinh(symbol)/cosh(symbol)
```

Comment:
Why not just return the integral of tanh here?

Reply:
It may not be a trivial thing for all people I guess.
IMHO `integrate(tanh(x))=log(cosh(x))` is not "atomic" enough. Atomic (or basic) integration rules come directly from basic derivation rules. We memorize `diff(sin(x))=cos(x)`, so we automatically get `integrate(cos(x))=sin(x)`. I don't think people need to memorize `diff(log(cosh(x))=tanh(x)`.

Reply:
Is there a more direct way to indicate what the next step should be then?

I don't know manualintegrate so well but it looks like `RewriteRule` goes back to the beginning and tries to figure out what rule to use again. Here at this stage we already know exactly what rules should be applied to compute the integral fully so that seems wasteful.

Reply:
That's a good point. Hard-coding further rules make code more complicated, but I do think the test cases I add run a bit slow. I will update it and do benchmark to see if it's worth.

Reply:
The `manualintegrate` routine is very slow and can often be the slowest part of a call to `integrate` so it makes general integration slow as well.

Reply:
I do agree that manualintegrate should aim to give user understandable steps and not just to produce an answer, since the steps can be used for things like "integration steps" in SymPy Gamma and similar tools. 

If the API were to allow specifying which rule to apply that would also mean it would be possible for the user to specify the rules directly instead of just having manualintegrate guess.

Reply:
I think it's possible here to specify the next step directly rather than asking `integral_steps` to choose the next step.